packages/navigator/src/browser/navigator-widget.ts

```diff
@@ -94,7 +94,7 @@ export class FileNavigatorWidget extends FileTreeWidget {
                 const exists = await this.fileSystem.exists(id);
                 if (exists) {
                     const fileStat = await this.fileSystem.getFileStat(id);
-                    if (!fileStat.isDirectory) {
+                    if (fileStat && fileStat.isDirectory) {
```

Comment:
First off, no need to check the file existence. The old code was executed when the resource existed, and it was not a directory, a file.

The current code executes if the file exists and a directory. I think this is incorrect; perhaps I am just overlooking something.

Reply:
No you're right thanks for looking at that, I got confused with `filestat!.isDirectory` I believe, this really changes the behavior.